app/src/main/java/com/example/androidtrainingproject/ui/home/HomeViewModel.kt

```diff
@@ -51,11 +57,52 @@ class HomeViewModel @Inject constructor(
         val newFilteredProducts = allProducts.filter { product: ProductResponse ->
             product.title.contains(newValue, ignoreCase = true)
         }
-        filteredProducts = newFilteredProducts
+        filteredProducts.clear()
+        filteredProducts.addAll(newFilteredProducts)
     }
 
     fun clearSearchText() {
         _searchText.value = ""
-        filteredProducts = allProducts
+        filteredProducts.clear()
+        filteredProducts.addAll(allProducts)
+    }
+
+    fun applyFilters(
+        checkedCategories: MutableList<String>,
+        rating: Int,
+        price: ClosedFloatingPointRange<Float>
+    ) {
+        filteredProducts.clear()
+        filteredProducts.addAll(allProducts.filter { product ->
```

Comment:
I'd personally place the whole `allProducts.filter {` or each of its 3 conditions into a custom getter/s to make all this a bit more lightweight and separated here.